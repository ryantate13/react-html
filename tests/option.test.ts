import {option} from '../src';

describe('option', () => {
    it('renders with children', () => {
        expect(option({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(option({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(option({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(option()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(option({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(option({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(option({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(option({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(option({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(option({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(option({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(option({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(option({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(option({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(option({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(option({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(option({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(option({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(option({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(option({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(option({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(option({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(option({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(option({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(option({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(option({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(option({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(option({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(option({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(option({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(option({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(option({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(option({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(option({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(option({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(option({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(option({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(option({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(option({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(option({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(option({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(option({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(option({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(option({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(option({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(option({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(option({translate: "no"})).toMatchSnapshot();
    });

    it('renders with disabled=true', () => {
        expect(option({disabled: true})).toMatchSnapshot();
    });

    it('renders with disabled=false', () => {
        expect(option({disabled: false})).toMatchSnapshot();
    });

    it('renders with label="TEST"', () => {
        expect(option({label: "TEST"})).toMatchSnapshot();
    });

    it('renders with selected=true', () => {
        expect(option({selected: true})).toMatchSnapshot();
    });

    it('renders with selected=false', () => {
        expect(option({selected: false})).toMatchSnapshot();
    });

    it('renders with value="TEST"', () => {
        expect(option({value: "TEST"})).toMatchSnapshot();
    });
});