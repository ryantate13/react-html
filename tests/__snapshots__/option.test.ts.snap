// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`option renders with accessKey="a,b,c,d" 1`] = `
<option
  accessKey="a,b,c,d"
/>
`;

exports[`option renders with autoCapitalize="characters" 1`] = `
<option
  autoCapitalize="characters"
/>
`;

exports[`option renders with autoCapitalize="none" 1`] = `
<option
  autoCapitalize="none"
/>
`;

exports[`option renders with autoCapitalize="off" 1`] = `
<option
  autoCapitalize="off"
/>
`;

exports[`option renders with autoCapitalize="on" 1`] = `
<option
  autoCapitalize="on"
/>
`;

exports[`option renders with autoCapitalize="sentences" 1`] = `
<option
  autoCapitalize="sentences"
/>
`;

exports[`option renders with autoCapitalize="words" 1`] = `
<option
  autoCapitalize="words"
/>
`;

exports[`option renders with children 1`] = `
<option>
  TEST
</option>
`;

exports[`option renders with className="foo bar baz" 1`] = `
<option
  className="foo bar baz"
/>
`;

exports[`option renders with contentEditable="false" 1`] = `
<option
  contentEditable="false"
/>
`;

exports[`option renders with contentEditable="true" 1`] = `
<option
  contentEditable="true"
/>
`;

exports[`option renders with dir="auto" 1`] = `
<option
  dir="auto"
/>
`;

exports[`option renders with dir="ltr" 1`] = `
<option
  dir="ltr"
/>
`;

exports[`option renders with dir="rtl" 1`] = `
<option
  dir="rtl"
/>
`;

exports[`option renders with disabled="false" 1`] = `
<option
  disabled={false}
/>
`;

exports[`option renders with disabled="true" 1`] = `
<option
  disabled={true}
/>
`;

exports[`option renders with draggable="false" 1`] = `
<option
  draggable="false"
/>
`;

exports[`option renders with draggable="true" 1`] = `
<option
  draggable="true"
/>
`;

exports[`option renders with empty attributes 1`] = `<option />`;

exports[`option renders with hidden="false" 1`] = `
<option
  hidden={false}
/>
`;

exports[`option renders with hidden="true" 1`] = `
<option
  hidden={true}
/>
`;

exports[`option renders with id="TEST" 1`] = `
<option
  id="TEST"
/>
`;

exports[`option renders with inputMode="decimal" 1`] = `
<option
  inputMode="decimal"
/>
`;

exports[`option renders with inputMode="email" 1`] = `
<option
  inputMode="email"
/>
`;

exports[`option renders with inputMode="none" 1`] = `
<option
  inputMode="none"
/>
`;

exports[`option renders with inputMode="numeric" 1`] = `
<option
  inputMode="numeric"
/>
`;

exports[`option renders with inputMode="search" 1`] = `
<option
  inputMode="search"
/>
`;

exports[`option renders with inputMode="tel" 1`] = `
<option
  inputMode="tel"
/>
`;

exports[`option renders with inputMode="text" 1`] = `
<option
  inputMode="text"
/>
`;

exports[`option renders with inputMode="url" 1`] = `
<option
  inputMode="url"
/>
`;

exports[`option renders with is="foo" 1`] = `
<option
  is="foo"
/>
`;

exports[`option renders with itemID="https://google.com/" 1`] = `
<option
  itemID="https://google.com/"
/>
`;

exports[`option renders with itemProp="https://google.com/" 1`] = `
<option
  itemProp="https://google.com/"
/>
`;

exports[`option renders with itemRef="foo bar" 1`] = `
<option
  itemRef="foo bar"
/>
`;

exports[`option renders with itemScope="false" 1`] = `
<option
  itemScope={false}
/>
`;

exports[`option renders with itemScope="true" 1`] = `
<option
  itemScope={true}
/>
`;

exports[`option renders with itemType="https://google.com/" 1`] = `
<option
  itemType="https://google.com/"
/>
`;

exports[`option renders with label="TEST" 1`] = `
<option
  label="TEST"
/>
`;

exports[`option renders with lang="" 1`] = `
<option
  lang=""
/>
`;

exports[`option renders with lang="es-419" 1`] = `
<option
  lang="es-419"
/>
`;

exports[`option renders with selected="false" 1`] = `
<option
  selected={false}
/>
`;

exports[`option renders with selected="true" 1`] = `
<option
  selected={true}
/>
`;

exports[`option renders with slot="TEST" 1`] = `
<option
  slot="TEST"
/>
`;

exports[`option renders with spellCheck="false" 1`] = `
<option
  spellCheck="false"
/>
`;

exports[`option renders with spellCheck="true" 1`] = `
<option
  spellCheck="true"
/>
`;

exports[`option renders with style="[object Object]" 1`] = `
<option
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`option renders with tabIndex="1" 1`] = `
<option
  tabIndex={1}
/>
`;

exports[`option renders with title="TEST" 1`] = `
<option
  title="TEST"
/>
`;

exports[`option renders with translate="no" 1`] = `
<option
  translate="no"
/>
`;

exports[`option renders with translate="yes" 1`] = `
<option
  translate="yes"
/>
`;

exports[`option renders with value="TEST" 1`] = `
<option
  value="TEST"
/>
`;

exports[`option renders without attributes 1`] = `<option />`;
